### ISSUE TITLE
[BUG]: v2 predict CLI does not work with multitask output - multitask MVE

### ISSUE THREAD

**Issue by davidegraff** [chemprop/chemprop#642]
**Describe the bug**
[These lines](https://github.com/chemprop/chemprop/blob/b7d33e72bd44523117cc4672682356434a0865cf/chemprop/cli/predict.py#L257-L259) will not work for a `preds` of shape $n \times t$ for $t \gt 1$:

```python
    df_test[
        "preds"
    ] = preds.flatten()
```

**Suggested fix**
Change it to:
```python
    df_test[
        "preds"
    ] = preds.tolist()
```

**Discussion**
It would be nice if we could output each task/target in a separate column. Consider an MPNN trained to predict LogP and QED. Ideally, the output CSV would look like:
```
smiles,logp,qed
CCCC,1.11,2.22
[...]
```
but with this fix it'll just look like:
```
smiles,preds
CCCC,[1.11, 2.22]
```
At the very least, this will work, but we should provide some mechanism for the prediction script to know what each column preds corresponds to. But that's work for a future PR.

**Comment by hwpang** [comment]
Working on this right now. A problem of having the appropriate column header is that it is possible for the test smiles input to only contain a column of smiles, and from the model architecture we know the number of task, but we wouldn't know the appropriate header.

**Comment by davidegraff** [comment]
I had the same idea. It could get quite verbose on the command line, but we require users to do this already in `chemprop train`, so I don't think it's that bad actually. The only alternative is to store this metadata somewhere inside the output directory of a training run, but I definitely prefer the former. One complication of labelling output columns in multi-target and multi-class outputs. That is, we might train a three-task MVE model ($\hat Y \in \mathbb R^{n \times 3 \times 2}$) or a two-task multiclass classification model ($\hat Y \in \mathbb R^{n \times 2 \times k}$), so we'd need to think about how to structure the output CSV properly.

**Comment by hwpang** [comment]
Base case multitask and multiclass multitask is addressed by https://github.com/chemprop/chemprop/pull/647. The multitask MVE is in the scope of v2.1. Moving this to v2.1 milestone.

**Comment by shihchengli** [comment]
The multitask MVE will be addressed via #959
